v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -2,6 +2,131 @@
+│   │   ├── Protocol: *Enum (2 values)
+│   │   │   ├── "http"
+│   │   │   └── "soap"
+│   │   ├── Proxy: *Object (3 properties)
+│   │   │   ├── Password: *genruntime.SecretReference
+│   │   │   ├── Url: Validated<*string> (2 rules)
+│   │   │   │   ├── Rule 0: MaxLength: 2000
+│   │   │   │   └── Rule 1: MinLength: 1
+│   │   │   └── Username: *string
+│   │   ├── ResourceReference: *genruntime.ResourceReference
+│   │   ├── Title: Validated<*string> (2 rules)
+│   │   │   ├── Rule 0: MaxLength: 300
+│   │   │   └── Rule 1: MinLength: 1
+│   │   ├── Tls: *Object (2 properties)
+│   │   │   ├── ValidateCertificateChain: *bool
+│   │   │   └── ValidateCertificateName: *bool
+│   │   └── Url: Validated<*string> (2 rules)
+│   │       ├── Rule 0: MaxLength: 2000
+│   │       └── Rule 1: MinLength: 1
+│   └── Status: Object (13 properties)
+│       ├── Conditions: conditions.Condition[]
+│       ├── Credentials: *Object (5 properties)
+│       │   ├── Authorization: *Object (2 properties)
+│       │   │   ├── Parameter: *string
+│       │   │   └── Scheme: *string
+│       │   ├── Certificate: string[]
+│       │   ├── CertificateIds: string[]
+│       │   ├── Header: map[string]string[]
+│       │   └── Query: map[string]string[]
+│       ├── Description: *string
+│       ├── Id: *string
+│       ├── Name: *string
+│       ├── Properties: *Object (1 property)
+│       │   └── ServiceFabricCluster: *Object (6 properties)
+│       │       ├── ClientCertificateId: *string
+│       │       ├── ClientCertificatethumbprint: *string
+│       │       ├── ManagementEndpoints: string[]
+│       │       ├── MaxPartitionResolutionRetries: *int
+│       │       ├── ServerCertificateThumbprints: string[]
+│       │       └── ServerX509Names: Object (2 properties)[]
+│       │           ├── IssuerCertificateThumbprint: *string
+│       │           └── Name: *string
+│       ├── Protocol: *Enum (2 values)
+│       │   ├── "http"
+│       │   └── "soap"
+│       ├── Proxy: *Object (2 properties)
+│       │   ├── Url: *string
+│       │   └── Username: *string
+│       ├── ResourceId: *string
+│       ├── Title: *string
+│       ├── Tls: *Object (2 properties)
+│       │   ├── ValidateCertificateChain: *bool
+│       │   └── ValidateCertificateName: *bool
+│       ├── Type: *string
+│       └── Url: *string
+├── NamedValue: Resource
+│   ├── Owner: Service
+│   ├── Spec: Object (7 properties)
+│   │   ├── AzureName: Validated<string> (2 rules)
+│   │   │   ├── Rule 0: MaxLength: 256
+│   │   │   └── Rule 1: Pattern: "^[^*#&+:<>?]+$"
+│   │   ├── DisplayName: Validated<*string> (3 rules)
+│   │   │   ├── Rule 0: MaxLength: 256
+│   │   │   ├── Rule 1: MinLength: 1
+│   │   │   └── Rule 2: Pattern: "^[A-Za-z0-9-._]+$"
+│   │   ├── KeyVault: *Object (3 properties)
+│   │   │   ├── IdentityClientId: *string
+│   │   │   ├── IdentityClientIdFromConfig: *genruntime.ConfigMapReference
+│   │   │   └── SecretIdentifierReference: *genruntime.ResourceReference
```

Comment:
Do you know if this value is actually supposed to be an ARM ID? I'm curious because sometimes RPs w/ KeyVault expect a KeyVault URL which is actually _not_ an ARM ID here, and we don't want to suggest it must be an ARM ID if it doesn't really need to be.

Reply:
I think it's this: https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.apimanagement/api-management-key-vault-create/azuredeploy.json#L162

Which is not actually an ARM ID, AFAIK

Reply:
This is for NamedValue. If I create one in the portal it is in the format `https://kv-jhg12hg-dev.vault.azure.net/secrets/kvs-aikey`

Reply:
No longer an ARM reference